netshare/learn/setup_cross_chunks_data.py

```diff
@@ -0,0 +1,256 @@
+        gk_chunk = df_chunk.groupby(by=[m.column for m in session_key_config])
+        max_flow_lens.append(max(gk_chunk.size().values))
+
+    return max(max_flow_lens)
+
+
+def get_word2vec_model(df: pd.DataFrame) -> Optional[Word2Vec]:
+    word2vec_config = get_config(
+        "pre_post_processor.config", path2="learn.word2vec", default_value={}
+    )
+    if not word2vec_config:
+        logger.debug("No learn.word2vec config found, skipping the embedding model")
+        return None
+    session_key_fields = word2vec_config.get(
+        "session_key", word2vec_config.get("metadata", [])
+    )
+    word2vec_cols = [
+        m
+        for m in (session_key_fields + word2vec_config.get("timeseries", []))
+        if "word2vec" in getattr(m, "encoding", "")
+    ]
+
+    if not word2vec_cols:
+        logger.info("Skipping word2vec embedding: no word2vec columns")
+        return None
+
+    if word2vec_config["word2vec"]["pretrain_model_path"]:
+        logger.info("Word2vec: Loading pretrained model")
+        return Word2Vec.load(word2vec_config["word2vec"]["pretrain_model_path"])
+
+    logger.info("Word2vec: training model")
+    os.makedirs(get_preprocessed_data_folder(), exist_ok=True)
+    word2vec_model_path = word2vec_train(
+        df=df,
+        out_dir=get_word2vec_model_directory(),
+        model_name=word2vec_config["word2vec"]["model_name"],
+        word2vec_cols=word2vec_cols,
+        word2vec_size=word2vec_config["word2vec"]["vec_size"],
+        annoy_n_trees=word2vec_config["word2vec"]["annoy_n_trees"],
+    )
+    return Word2Vec.load(word2vec_model_path)
+
+
+def build_field_from_config(field: Config, df: pd.DataFrame) -> Field:
+    """
+    This function builds the Field object from a specific field configuration.
+    There are 3 types of fields:
+    1. BitField: when encoding = bit. This field using the configuration n_bits.
+    2. ContinuousField:
+        2.1 when encoding = word2vec, we use the word2vec model to encode the field.
+        2.2 when type = float, we take the min and max value of the column in the dataframe.
+    3. DiscreteField: when encoding = categorical or type = string, we use the unique values of the column in the dataframe.
+    """
+    prepost_config = get_config("pre_post_processor.config", default_value={})
+
+    if not isinstance(field.get("column"), str) and not isinstance(
+        field.get("columns"), list
+    ):
+        raise ValueError(
+            f'In field configuration: "column" should be a string or "columns" a list of strings ({field})'
+        )
+    if "type" not in field or field.type not in get_config(
+        "global_config.allowed_data_types", default_value=[field.type]
+    ):
+        raise ValueError(
+            '"type" must be specified as ({})'.format(
+                " | ".join(get_config("global_config.allowed_data_types"))
+            )
+        )
+
+    field_name = getattr(field, "name", "") or field.get("column", "") or field.columns
```

Comment:
they are actually different: `getattr` aimed to grab an attribute from an object, and `get` grabs a key from a dictionary!
This is `Config`'s dark magic that supports both of them 😂 
I'll change to `.get`, so it will be consistent as you suggested :)

Reply:
I'll also change things like `field.type` to `field["type"]`. I think that it also helps the reader understand that there's no complex object here, only fetching data from the config.